hey_fireball.py

```diff
@@ -149,6 +151,29 @@ def _extract_count(self):
                     return int(self.parts[idx])
                 except ValueError:
                     pass
+
+    def _extract_setting(self):
+        if self.bot_is_first:
+            idx = 2
+            curPref = get_pm_preference(self.requestor_id)
+            try:
+                self.parts[idx]
```

Comment:
Instead of a try-except, I would check the length of self.parts to determine if which logic branch to execute. Typically, try-except is only using for flow control like this if the statement because caught does something (adds a value to a collection, gets next value in a collection, etc.), but `self.parts[idx]` does not serve a purpose other than to determine if idx=2 is value.

Reply:
try-except has been removed. Now we're checking if self.parts < 3